controllers/adapters_controller.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Adapter Controller.
+ *
+ * Manages HTTP requests to /adapters. 
+ *  
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+var express = require('express');
+var adapterManager = require('../adapter-manager');
+
+var adaptersController = express.Router();
+
+/**
+ * Return a list of adapters
+ */
+adaptersController.get('/', (request, response) => {
+  adapterManager.discoverAdapters().then(function(adapters) {
+    var adapterList = [];
+    for (var adapter of adapters) {
+      adapterList.push(adapter.asDict());
+    }
+    response.json(adapterList);
+  });
+});
+
+/**
+ * Cancel Action: add a new device to the adapter.
+ */
+adaptersController.delete('/addSomeDevice', (request, response) => {
```

Comment:
Well the action in this case is to initiate pairing. Since there can only be one pairing request I chose not to bother implementing a queue and returned an href which is the same as the GET. So you do a GET on /adapters/addSomeDevice to initiate pairing and you do a DELETE on /adapters/addSomeDevice to cancel the pairing action.

I could fabricate some dynamic actions stuff but that seemed like an awful lot of "make work" just for a debug feature.